From sean@mess.org  Wed Dec  6 12:52:51 2006
Date: Wed, 6 Dec 2006 20:27:32 +0000
From: Sean Young <sean@mess.org>
To: Greg KH <gregkh@suse.de>, Chris Wright <chrisw@sous-sol.org>
Cc: linux-kernel@vger.kernel.org, linux-usb-devel@lists.sourceforge.net
Subject: USB: Fix oops in PhidgetServo
Message-ID: <20061206202732.GA91199@atlantis.8hz.com>

From: Sean Young <sean@mess.org>

The PhidgetServo causes an Oops when any of its sysfs attributes are read
or written too, making the driver useless.

Signed-off-by: Sean Young <sean@mess.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/phidgetservo.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/usb/misc/phidgetservo.c
+++ gregkh-2.6/drivers/usb/misc/phidgetservo.c
@@ -282,6 +282,7 @@ servo_probe(struct usb_interface *interf
 		dev->dev = NULL;
 		goto out;
 	}
+	dev_set_drvdata(dev->dev, dev);
 
 	servo_count = dev->type & SERVO_COUNT_QUAD ? 4 : 1;
 
